File: Algorithms/Easy/making-anagrams.py
#!/bin/python3

import math
import os
import random
import re
import sys

"""
@url: https://www.hackerrank.com/challenges/making-anagrams
@desc: 可以用map和lambda
"""

# Complete the palindromeIndex function below.
def makingAnagrams(s1, s2):
    return sum(list(map(lambda x: abs(s1.count(x) - s2.count(x)), [chr(n) for n in range(97, 123)])))

File: Algorithms/Easy/making-anagrams.php
<?php

/**
 * @desc 计算两边的差集
 * @desc https://www.hackerrank.com/challenges/making-anagrams
 */
function makingAnagrams($s1, $s2) {
    $deletions = 0;
    for ($i=97; $i<=122; $i++) {
        $c = chr($i);
        $deletions += abs(substr_count($s1, $c) - substr_count($s2, $c));
    }
    return $deletions;
}
